src/ast/semantic_analyser.cpp

```diff
@@ -1594,6 +1594,68 @@ void SemanticAnalyser::binop_int(Binop &binop)
   }
 }
 
+void SemanticAnalyser::binop_ptr(Binop &binop)
+{
+  auto &lht = binop.left->type;
+  auto &rht = binop.right->type;
+
+  bool left_is_ptr = lht.IsPtrTy();
+  auto &ptr = left_is_ptr ? lht : rht;
+  auto &other = left_is_ptr ? rht : lht;
+
+  auto compare = false;
+
+  // Do what C does
+  switch (binop.op)
+  {
+    case bpftrace::Parser::token::EQ:
+    case bpftrace::Parser::token::NE:
+    case bpftrace::Parser::token::LE:
+    case bpftrace::Parser::token::GE:
+    case bpftrace::Parser::token::LT:
+    case bpftrace::Parser::token::GT:
+      compare = true;
+      break;
+    default:;
+  }
+
+  auto invalid_op = [&binop, this, &lht, &rht]() {
+    LOG(ERROR, binop.loc, err_)
+        << "The " << opstr(binop)
+        << " operator can not be used on expressions of types " << lht << ", "
+        << rht;
+  };
+
+  // Binop on two pointers
+  if (other.IsPtrTy())
+  {
+    if (compare)
```

Comment:
Clang emits a warning, we should probably copy it

test.c:6:8: warning: comparison of distinct pointer types ('char *' and 'int *')
      [-Wcompare-distinct-pointer-types]